### PR TITLE
Add shape inference for clip

### PR DESCRIPTION
**Pull request by Zehaos**

### CODE REVIEW

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 1]
```diff
@@ -3244,7 +3244,16 @@ LogicalResult ONNXCeilOp::inferShapes() {
 }
 
 LogicalResult ONNXClipOp::inferShapes() {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  if (!input().getType().isa<RankedTensorType>())
+    return emitError("Input tensor not ranked");
+
+  auto inputTy = input().getType().cast<RankedTensorType>();
```
Comment (AlexandreEichenberger) [inline src/Dialect/ONNX/ONNXOps.cpp:3250]: nit: please do not use auto unless the types are very complicated.

**Review by AlexandreEichenberger — COMMENTED** [review]
Please changes the auto to normal type, otherwise everything looks good, thanks

**Comment by doru1004** [comment]
@jenkins-droid test this please

**Comment by Zehaos** [comment]
@AlexandreEichenberger  done.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, pending upgrading to the latest ONNX-MLIR and passing the tests. Thanks

**Comment by tjingrant** [comment]
@jenkins-droid test this please.

**Comment by tjingrant** [comment]
@jenkins-droid test this please.

**Pull request merged**